src/test/java/com/varukha/onlinebookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,212 @@
+        USER.setShippingAddress("15 Main St, City, Country");
+        USER.setDeleted(false);
+
+        VALID_CATEGORY_1.setId(1L);
+        VALID_CATEGORY_1.setName("Science Fiction");
+        VALID_CATEGORY_1.setDescription("Science Fiction category description");
+
+        BOOK.setTitle("Sample Book Title 1");
+        BOOK.setAuthor("Author 1");
+        BOOK.setIsbn("978-1234567891");
+        BOOK.setPrice(BigDecimal.valueOf(25));
+        BOOK.setDescription("Description 1");
+        BOOK.setCoverImage("https://example.com/book-cover1.jpg");
+        BOOK.setCategories(Set.of(VALID_CATEGORY_1));
+
+        VALID_CART_ITEM_REQUEST.setBookId(1L);
+        VALID_CART_ITEM_REQUEST.setQuantity(1);
+
+        VALID_CART_ITEM_RESPONSE.setId(1L);
+        VALID_CART_ITEM_RESPONSE.setBookId(BOOK.getId());
+        VALID_CART_ITEM_RESPONSE.setQuantity(VALID_CART_ITEM_REQUEST.getQuantity());
+        VALID_CART_ITEM_RESPONSE.setBookTitle(BOOK.getTitle());
+
+        CART_ITEM.setId(1L);
+        CART_ITEM.setShoppingCart(SHOPPING_CART);
+        CART_ITEM.setBook(BOOK);
+        CART_ITEM.setQuantity(1);
+
+        SHOPPING_CART.setId(1L);
+        SHOPPING_CART.setUser(USER);
+        SHOPPING_CART.setCartItems(Set.of());
+
+        VALID_SHOPPING_CART_DTO.setUserId(1L);
+        VALID_SHOPPING_CART_DTO.setCartItems(Set.of(VALID_CART_ITEM_RESPONSE));
+
+        VALID_UPDATE_REQUEST.setQuantity(5);
+
+        VALID_UPDATE_RESPONSE.setId(VALID_CART_ITEM_RESPONSE.getId());
+        VALID_UPDATE_RESPONSE.setBookId(VALID_CART_ITEM_RESPONSE.getBookId());
+        VALID_UPDATE_RESPONSE.setQuantity(VALID_UPDATE_REQUEST.getQuantity());
+        VALID_UPDATE_RESPONSE.setBookTitle(VALID_CART_ITEM_RESPONSE.getBookTitle());
+    }
+
+    @Test
+    @WithMockUser(username = "alice@gmail.com")
+    @DisplayName("""
+            Test the 'saveCartItem' endpoint for saving new cart item
+            to user shopping cart with valid request parameters
+             """)
+    void saveCartItem_ValidRequestData_ReturnCartItemDto() throws Exception {
+        String jsonRequest = objectMapper.writeValueAsString(VALID_CART_ITEM_REQUEST);
+        MvcResult result = mockMvc.perform(post("/carts")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+        CartItemDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                CartItemDto.class
+        );
+        assertNotNull(actual);
+        EqualsBuilder.reflectionEquals(
+                VALID_CART_ITEM_RESPONSE,
+                actual);
+    }
+
+    @Test
+    @WithMockUser(username = "alice@gmail.com")
+    @DisplayName("""
+            Test the 'deleteCartItemById' endpoint to delete
+            cart item from shopping cart by valid cart item ID
+             """)
+    void deleteCartItemById() throws Exception {
```

Comment:
It's better to rename this test to something like that - deleteCartItemByValidId_ResponseStatusNoContent()